Upstream pythoncapi-compat dropped inclusion of this header in
https://github.com/python/pythoncapi-compat/commit/11cb80f2652cb2fe5231bf60b9dd98c83a4e25f4
and pygobject upstream haven't yet followed suite.

diff --git a/gi/pygi-async.c b/gi/pygi-async.c
index 09829ed..c0d63be 100644
--- a/gi/pygi-async.c
+++ b/gi/pygi-async.c
@@ -24,6 +24,8 @@
 #include "pygi-invoke.h"
 #include "pygi-util.h"
 
+#include "structmember.h"
+
 static PyObject *asyncio_InvalidStateError;
 static PyObject *asyncio_get_running_loop;
 #if defined(PYPY_VERSION)
